src/main.rs

```diff
@@ -494,63 +732,121 @@ fn get_setup(args: &[String]) -> Setup {
             }),
         };
 
+        #[cfg(not(feature = "alsa-backend"))]
+        let mixer_device = MixerConfig::default().device;
+
+        #[cfg(feature = "alsa-backend")]
         let index = match matches.opt_str("mixer-index") {
             Some(index) => {
-                warn!("--mixer-index is deprecated and will be removed in a future release.");
-                warn!("Please use --alsa-mixer-index instead.");
-                index
-                    .parse::<u32>()
-                    .expect("Mixer index is not a valid number")
+                warn!("`--mixer-index` is deprecated and will be removed in a future release.");
+                warn!(
+                    "Please use `--{}` / `-{}`.",
+                    ALSA_MIXER_INDEX, ALSA_MIXER_INDEX_SHORT
+                );
+
+                index.parse::<u32>().unwrap_or_else(|_| {
+                    error!(
+                        "Invalid `--{}` / `-{}`: {}",
+                        ALSA_MIXER_INDEX, ALSA_MIXER_INDEX_SHORT, index
+                    );
+                    println!("Default: {}", MixerConfig::default().index);
+                    exit(1);
+                })
             }
             None => matches
                 .opt_str(ALSA_MIXER_INDEX)
                 .map(|index| {
-                    index
-                        .parse::<u32>()
-                        .expect("Alsa mixer index is not a valid number")
+                    index.parse::<u32>().unwrap_or_else(|_| {
+                        error!(
+                            "Invalid `--{}` / `-{}`: {}",
+                            ALSA_MIXER_INDEX, ALSA_MIXER_INDEX_SHORT, index
+                        );
+                        println!("Default: {}", MixerConfig::default().index);
+                        exit(1);
+                    })
                 })
-                .unwrap_or(0),
+                .unwrap_or_else(|| MixerConfig::default().index),
```

Comment:
`unwrap_or` might suffice here, because that default index is a constant.

Reply:
It may not always be in the future though?

Reply:
In the future who knows the logic to find `MixerConfig::default().index` might change and maybe it won't always be zero? I don't know. You're the mixer guy. Do you foresee that being a possibility?

Reply:
I also dried up the `ConnectConfig::default()` stuff.

Reply:
`MixerConfig::default()` is also dried up too.